packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -544,8 +548,37 @@ export class ProjectService {
         return this.jobModel.getMostRecentJobByProject(projectUuid);
     }
 
-    async getJobStatus(jobUuid: string): Promise<Job> {
-        return this.jobModel.get(jobUuid);
+    async getJobStatus(jobUuid: string, user: SessionUser): Promise<Job> {
+        const job = await this.jobModel.get(jobUuid);
+
+        if (job.projectUuid === undefined) {
+            if (job.userUuid !== user.userUuid) {
+                throw new AuthorizationError("User can't see this job");
+            }
+        } else {
+            const { cannot } = new AbilityBuilder<OrganizationMemberAbility>(
+                Ability,
+            );
+            if (
+                cannot('view', 'Project', {
+                    organisationUuid: user.organizationUuid,
+                })
+            ) {
+                // throw new AuthorizationError("User has no access to this project");
```

Comment:
@owlas am I doing anything wrong with this AbilityBuilder ? if I uncommet this, then the test  `should get job with projectUuid if user belongs to org`  will break

Reply:
In the `else` block you're defining a new ability and not using it. Later we only check `ability`.

I think this extra logic needs to live in the `common` package with the other abilities.

Give viewers the ability to view all jobs created by their user